### PR TITLE
Remove disco react warnings

### LINKED ISSUE #6059: Cleanup errors/warnings when running `yarn disco`
There are a lot of errors related to invalid props when starting disco locally. We should fix those errors. See #5860 for amo.

For QA:

- the install button should continue to work in -stage (which is not the same button as in -dev)
- the disco pane should work as intended in -dev (i.e. installing/removing add-ons with the mozAddonManager)

### PR DESCRIPTION
**Pull request by willdurand**
Fix #6059

---

This PR removes react warnings in the disco app, mostly by fixing invalid prop types.

### CODE REVIEW

File: src/core/addonManager.js [hunk 1]
```diff
@@ -8,14 +8,11 @@ import {
...
 
-const testHasWindow = () => typeof window !== 'undefined';
-
-export function hasAddonManager({ hasWindow = testHasWindow, navigator } = {}) {
-  // Returns undefined if it cannot be determined if mozAddonManager is supported (likely server
-  // rendering with no window). Otherwise returns true/false based on mozAddonManager in navigator.
-  if (!navigator && !hasWindow()) {
-    return undefined;
+export function hasAddonManager({ navigator } = {}) {
+  if (typeof window === 'undefined') {
```
Comment (rebmullin) [inline src/core/addonManager.js]: should we still do a check for navigator? is it possible it won't be defined either?

File: tests/unit/disco/components/TestDiscoPane.js [hunk 2]
```diff
@@ -327,12 +319,25 @@ describe(__filename, () => {
   });
 
   describe('componentDidMount', () => {
-    it('sets events', () => {
+    it('does not set events on the server', () => {
```
Comment (rebmullin) [inline tests/unit/disco/components/TestDiscoPane.js]: I guess this ties in to your above change but still not clear why this change is needed..

File: src/disco/components/DiscoPane/index.js [hunk 3]
```diff
@@ -81,9 +83,15 @@ export class DiscoPaneBase extends React.Component {
   componentDidMount() {
     const {
       _addChangeListeners,
+      _config,
       handleGlobalEvent,
       mozAddonManager,
     } = this.props;
+
+    if (_config.get('server')) {
```
Comment (rebmullin) [inline src/disco/components/DiscoPane/index.js]: I'm just curious what issue is this fixing...

File: src/core/addonManager.js (hunk 1 above)
Comment (willdurand) [reply src/core/addonManager.js]: It does not matter at this point, because the next condition will be false.

File: tests/unit/disco/components/TestDiscoPane.js (hunk 2 above)
Comment (willdurand) [reply tests/unit/disco/components/TestDiscoPane.js]: because it was the same result before, but very obscure. it

File: src/disco/components/DiscoPane/index.js (hunk 3 above)
Comment (willdurand) [reply src/disco/components/DiscoPane/index.js]: it does not fix an issue but make the code more readable.

Comment (willdurand) [reply src/disco/components/DiscoPane/index.js]: it is related to your previous question/my previous answer.

File: src/core/addonManager.js (hunk 1 above)
Comment (rebmullin) [reply src/core/addonManager.js]: 🤦‍♂️ yep I see : )

File: src/disco/components/DiscoPane/index.js (hunk 3 above)
Comment (rebmullin) [reply src/disco/components/DiscoPane/index.js]: is this even needed tho? I thought componentDidMount lifecycle was only called on client?

File: src/disco/components/DiscoPane/index.js [hunk 4]
```diff
@@ -38,13 +38,15 @@ export class DiscoPaneBase extends React.Component {
...
     _addChangeListeners: PropTypes.func,
+    _config: PropTypes.object,
     _tracking: PropTypes.object,
   };
 
   static defaultProps = {
     AddonComponent: Addon,
     mozAddonManager: config.get('server') ? {} : navigator.mozAddonManager,
     _addChangeListeners: addChangeListeners,
+    _config: config,
```
Comment (rebmullin) [inline src/disco/components/DiscoPane/index.js]: just checking..do we still need this?

Comment (willdurand) [reply src/disco/components/DiscoPane/index.js]: ah nope

Comment (willdurand) [reply src/disco/components/DiscoPane/index.js]: removed!
